velox/dwio/common/TypeWithId.cpp

```diff
@@ -64,8 +64,8 @@ namespace {
 
 int countNodes(const TypePtr& type) {
   int count = 1;
-  for (auto& child : *type) {
-    count += countNodes(child);
+  for (uint32_t end = type->size(), i = 0; i < end; ++i) {
```

Comment:
`for (uint32_t end = type->size(), i = 0; i < end; ++i)` should have the same performance as `for (uint32_t i = 0; i < type->size(); ++i)`, where `end` stores in a register, right?